docs: ac mirror after install (#23943)
newrelic/docs-website: src/content/docs/new-relic-control/agent-control/mirror-setup.mdx

---
title: "Configure an OCI registry mirror for Agent Control"
metaDescription: "Learn how to configure Agent Control to use an OCI registry mirror."
freshnessValidatedDate: never
---
<Callout variant="important">
  Agent Control and New Relic Control are now **generally available** for Kubernetes! Support for Linux hosts and Windows hosts is also in **public preview** program, pursuant to our [pre-release policies](/docs/licenses/license-information/referenced-policies/new-relic-pre-release-policy).
</Callout>

> Agent Control uses `docker.io` as the default OCI (Open Container Initiative) registry for downloading agent packages. If your environment requires using a different registry, you can configure Agent Control to use a custom OCI registry or mirror.

Reviewer: ```suggestion
By default, Agent Control uses docker.io as the Open Container Initiative (OCI) registry to download agent packages. If your environment requires a different registry, you can configure a custom OCI registry or mirror.
```

## Before you begin

Before configuring a custom OCI registry, ensure you have:

> 1. **Installed Agent Control:** Complete the standard [Agent Control installation](/docs/new-relic-control/agent-control/setup) using the New Relic CLI

Reviewer: ```suggestion
1. **Installed Agent Control:** Use the New Relic CLI to complete the [Agent Control installation](/docs/new-relic-control/agent-control/setup).
```

> 2. **Mirror registry available:** Your custom OCI registry should be accessible from the host where Agent Control is installed

Reviewer: ```suggestion
2. **Mirror registry available:** Ensure your custom OCI registry is accessible from the host machine.
```

### Step 1: Edit the local configuration file

Edit the Agent Control local configuration file:

Linux - /etc/newrelic-agent-control/local-data/agent-control/local_config.yaml
Windows - C:\Program Files\New Relic\newrelic-agent-control\local-data\agent-control\local_config.yaml

Add the following configuration sections to the file:

```yaml
oci:
  registry: YOUR_REGISTRY_URL
```

If you need to set up your username and password, use the following snippet:

```yaml
oci:
  registry: YOUR_REGISTRY_URL
  auth:
    basic:
      username: YOUR_USERNAME
      password: YOUR_PASSWORD
```

Replace the placeholder values:
- `YOUR_REGISTRY_URL`: The URL of your custom OCI registry (for example, `registry.example.com:5000` or `mirror.internal.company.com`)
- `YOUR_USERNAME`: Username for registry authentication
- `YOUR_PASSWORD`: Password for registry authentication

This assumes the machine has access to the public keys of each agent and that the mirror contains all the signatures up to date.
If this cannot be met, we can disable signatures verification. We don't suggest it.

```yaml
agent_packages:
  signature_verification_enabled: false
```

> ### Step 3: Restart Agent Control

Reviewer: @danielorihuela, there is no step 2, are we missing something?  or just an incorrect numbering?

Author: Incorrect numbering. I started with three steps, collapsed them into two steps and forget to update the numbers.

Author: solved!!!
